api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -558,6 +521,33 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
             .workspace(workspaceMapper.toApiWorkspace(savedWorkspace, toFcWorkspace)));
   }
 
+  // A retry period is needed because the permission to copy files into the cloned workspace is not granted transactionally
+  private void copyBlobWithRetries(String bucketName, Blob b, int retryPeriodSeconds) {
+    final int SLEEP_INTERVAL_SECONDS = 5;
+
+    try {
+      cloudStorageService.copyBlob(b.getBlobId(), BlobId.of(bucketName, b.getName()));
+    } catch (StorageException e) {
+      if (retryPeriodSeconds <= 0) {
+        throw e;
+      }
+
+      log.info(
```

Comment:
I would think this should at least be a warning. Don't we try to make sure every account is provisioned with the right access programmatically? I.e. getting here is probably not just a user error or transient condition if I understand correctly.

Reply:
The permission grant is eventually consistent so reaching this code block is part of expected behavior. I'm ok with bumping it up to a warning though since a flood of these would indicate a problem.

Reply:
Sounds good.